### PR TITLE
feet: add zkera-finance adapter

### PR DESCRIPTION
**Pull request by zkeraFi**
Volume adapter for zkEra (https://defillama.com/protocol/zkera-finance)

### CODE REVIEW

**Comment by llamatester** [comment]
The zkera-finance adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts dexs zkera-finance
    
    🦙 Running ZKERA-FINANCE adapter 🦙
    ---------------------------------------------------
    Start Date:	Thu, 31 Oct 2024 09:36:46 GMT
    End Date:	Fri, 01 Nov 2024 09:36:46 GMT
    ---------------------------------------------------
    
    TELOS 👇
    Backfill start time: 31/7/2021
    NO METHODOLOGY SPECIFIED
    Daily volume: 9.45 k
    End timestamp: 1730453805 (2024-11-01T09:36:45.000Z)

File: dexs/zkera-finance/zkera.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import { request, gql } from "graphql-request";
+import { FetchOptions } from "../../adapters/types";
+import BigNumber from "bignumber.js";
+
+const graphUrl =
+  "https://api.goldsky.com/api/public/project_clzelu5d634f501x8ai8111wj/subgraphs/zlp-pool/latest/gn";
+const getData = async (timestamp: number) => {
+  const query = gql`
+    {
+      volumeStats(first: 1, where: { period: daily, timestamp: ${timestamp} }) {
```
Comment (dtmkeng) [inline dexs/zkera-finance/zkera.ts:10]: we count `
      liquidation
      margin` for volume could you fix ?

**Review by dtmkeng — CHANGES_REQUESTED** [review]

File: dexs/zkera-finance/zkera.ts (hunk 1 above)
Comment (g1nt0ki) [reply dexs/zkera-finance/zkera.ts:10]: also, how does this timestamp field work, it returns data for 24 hours before given timestamp or after?

Comment (zkeraFi) [reply dexs/zkera-finance/zkera.ts:10]: > we count ` liquidation margin` for volume could you fix ?

What do you mean? Can you explain?

Comment (zkeraFi) [reply dexs/zkera-finance/zkera.ts:10]: > also, how does this timestamp field work, it returns data for 24 hours before given timestamp or after?

returning data for the last 24h after timestamp 00:00 UTC

Comment (dtmkeng) [reply dexs/zkera-finance/zkera.ts:10]: this example https://github.com/DefiLlama/dimension-adapters/blob/master/dexs/gmx/index.ts#L22

Comment (zkeraFi) [reply dexs/zkera-finance/zkera.ts:10]: > this example https://github.com/DefiLlama/dimension-adapters/blob/master/dexs/gmx/index.ts#L22

volumeStats(where: { period: daily, timestamp: ${timestamp}}, first: 1) {
    timestamp
    mint
    burn
    margin
    swap
    liquidation
  }


there is separate field for liquidations in the graph

**Comment by llamatester** [comment]
The zkera-finance adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts dexs zkera-finance
    
    🦙 Running ZKERA-FINANCE adapter 🦙
    ---------------------------------------------------
    Start Date:	Wed, 06 Nov 2024 09:30:44 GMT
    End Date:	Thu, 07 Nov 2024 09:30:44 GMT
    ---------------------------------------------------
    
    TELOS 👇
    Backfill start time: 31/7/2021
    NO METHODOLOGY SPECIFIED
    Daily volume: 10.57 k
    End timestamp: 1730971843 (2024-11-07T09:30:43.000Z)

**Pull request merged**